lib/floe/workflow.rb

```diff
@@ -16,21 +16,71 @@ def load(path_or_io, context = nil, credentials = {}, name = nil)
         new(payload, context, credentials, name)
       end
 
-      def wait(workflows, timeout: nil)
+      def wait(workflows, timeout: nil, &block)
+        workflows = [workflows] if workflows.kind_of?(self)
         logger.info("checking #{workflows.count} workflows...")
 
-        start = Time.now.utc
-        ready = []
+        run_until   = Time.now.utc + timeout if timeout.to_i > 0
+        ready       = []
+        queue       = Queue.new
```

Comment:
Is there a concurrency concern here?  i.e. should we be using something from concurrent-ruby, or is Queue thread safe enough?

Reply:
Is `Queue` not thread-safe?  We use it in ManageIQ core across threads in a few places

Reply:
The docs for `Queue` show examples using multiple threads

Reply:
I can never remember hence my question - Some thing in Ruby are thread safe, others are partially and others are not.  Otherwise there wouldn't be a Concurrent::Hash :(